Managers/ActivityMan.cpp

```diff
@@ -1,1852 +1,164 @@
-    if (m_pControlledActor[player])
-    {
-        m_pControlledActor[player]->SetControllerMode(Controller::CIM_PLAYER, player);
-
-        // Play actor switching sound effects
-        // Brain has its own special sound effects
-        if (m_pControlledActor[player] == m_pBrain[player])
-            g_GUISound.BrainSwitchSound()->Play();
-        else
-            g_GUISound.ActorSwitchSound()->Play();
-        // Only play air swoosh if actors are out of sight of each other
-        if (pPrevActor && Vector(pPrevActor->GetPos() - m_pControlledActor[player]->GetPos()).GetMagnitude() > g_FrameMan.GetResX() / 2)
-            g_GUISound.CameraTravelSound()->Play();
-
-        ReassignSquadLeader(player, team);
-    }
-
-    // Follow the new guy normally
-    m_ViewState[player] = NORMAL;
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          SetDifficulty
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Sets the current difficulty setting
-
-void Activity::SetDifficulty(int difficulty)
-{
-    // Translate specific enums to be within their ranges, and not just at the limits
-    if (difficulty == CAKEDIFFICULTY)
-        m_Difficulty = 10;
-    else if (difficulty == EASYDIFFICULTY)
-        m_Difficulty = 30;
-    else if (difficulty == MEDIUMDIFFICULTY)
-        m_Difficulty = 50;
-    else if (difficulty == HARDDIFFICULTY)
-        m_Difficulty = 70;
-    else if (difficulty == NUTSDIFFICULTY)
-        m_Difficulty = 95;
-    else
-        m_Difficulty = difficulty;
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          IsPlayerTeam
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Indicates whether a team is player controlled or not.
-
-bool Activity::IsPlayerTeam(int team)
-{
-	if (team < Activity::TEAM_1 || team >= Activity::MAXTEAMCOUNT)
-		return false;
-
-    for (int player = 0; player < MAXPLAYERCOUNT; ++player)
-    {
-        if (m_IsActive[player] && m_Team[player] == team && m_IsHuman[player])
-            return true;
-    }
-
-    return false;
-}
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+	void ActivityMan::PauseActivity(bool pause) {
+		if (!m_Activity || (pause && m_Activity->Paused()) || (!pause && !m_Activity->Paused())) {
+			return;
+		}
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Virtual method:  SetupPlayers
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Precalculates the player-to-screen index map, counts the number of
-//                  active players, teams etc.
+		if (m_Activity) {		
+			if (pause) {
+				// Save the current in-game music position on pause
```

Comment:
I would rather these comments get subsumed into the method comment, i.e. something like
```
		/// <summary>
		/// Pauses and unpauses the game, saving and resuming in-game music if possible, or queuing default music if not.
		/// </summary>
		/// <param name="pause">Whether to pause the game or not.</param>
```